src/py2spack/loading.py

```diff
@@ -0,0 +1,266 @@
+"""Utils for downloading packages from PyPI and opening the pyproject.tomls."""
+
+import io
+import tarfile
+from typing import Any, Dict, List, Optional
```

Comment:
From python 3.9 builtin types like `dict` and `list` are generic and therefore they should be used directly, since their aliases in typing are deprecated (https://typing.readthedocs.io/en/latest/guides/modernizing.html#generics-in-the-typing-module)
```suggestion
from typing import Any, Optional
```